tests/datasource/fluent/data_asset/data_connector/test_filesystem_data_connector.py

```diff
@@ -832,73 +644,49 @@ def test_relative_base_directory_path(tmp_path_factory):
     assert len(my_batch_definition_list) == 1
 
 
-# TODO: <Alex>ALEX-UNCOMMENT_WHEN_SORTERS_ARE_INCLUDED_AND_TEST_SORTED_BATCH_DEFINITION_LIST</Alex>
-# TODO: <Alex>ALEX</Alex>
-# def test_return_all_batch_definitions_sorted_sorter_named_that_does_not_match_group(
-#     tmp_path_factory,
-# ):
-#     base_directory = str(
-#         tmp_path_factory.mktemp(
-#             "test_return_all_batch_definitions_sorted_sorter_named_that_does_not_match_group"
-#         )
-#     )
-#     create_files_in_directory(
-#         directory=base_directory,
-#         file_name_list=[
-#             "alex_20200809_1000.csv",
-#             "eugene_20200809_1500.csv",
-#             "james_20200811_1009.csv",
-#             "abe_20200809_1040.csv",
-#             "will_20200809_1002.csv",
-#             "james_20200713_1567.csv",
-#             "eugene_20201129_1900.csv",
-#             "will_20200810_1001.csv",
-#             "james_20200810_1003.csv",
-#             "alex_20200819_1300.csv",
-#         ],
-#     )
-#     my_data_connector_yaml = yaml.load(
-#         f"""
-#         class_name: FilesystemDataConnector
-#         datasource_name: test_environment
-#         base_directory: {base_directory}
-#         glob_directive: "*.csv"
-#         assets:
-#             my_filesystem_data_asset:
-#                 pattern: (.+)_(.+)_(.+)\\.csv
-#                 group_names:
-#                     - name
-#                     - timestamp
-#                     - price
-#         default_regex:
-#             pattern: (.+)_.+_.+\\.csv
-#             group_names:
-#                 - name
-#         sorters:
-#             - orderby: asc
-#               class_name: LexicographicSorter
-#               name: name
-#             - datetime_format: "%Y%m%d"
-#               orderby: desc
-#               class_name: DateTimeSorter
-#               name: timestamp
-#             - orderby: desc
-#               class_name: NumericSorter
-#               name: for_me_Me_Me
-#     """,
-#     )
-#     with pytest.raises(gx_exceptions.DataConnectorError):
-#         # noinspection PyUnusedLocal
-#         my_data_connector: FilesystemDataConnector = (
-#             instantiate_class_from_config(
-#                 config=my_data_connector_yaml,
-#                 runtime_environment={
-#                     "name": "fluent",
-#                     "execution_engine": PandasExecutionEngine(),
-#                 },
-#                 config_defaults={
-#                     "module_name": "great_expectations.datasource.data_connector"
-#                 },
-#             )
-#         )
-# TODO: <Alex>ALEX</Alex>
+@pytest.mark.filesystem
+@pytest.mark.parametrize(
+    "batching_regex,batch_definition_count",
+    [
+        pytest.param(MATCH_ALL_PATTERN, 6, id="Match All Pattern"),
+        pytest.param(r"alpha-(.*)\.csv", 3, id="Match Alpha CSV with Group"),
+        pytest.param(r"single-match\..*", 1, id="Match a single CSV"),
```

Comment:
Does it return an empty collection, or raise, when you pass in non-matching regex? Might be worth a test case, but I think this looks fantastic

Reply:
returns an empty list - [added test case](https://github.com/great-expectations/great_expectations/pull/9710/commits/8b1e2c35bfc4f20a980f653a72b0db7681e50570) to capture

Reply:
Thanks a billion